### PR TITLE
Two parallelized functions

### PR DESCRIPTION
**Pull request by damonge**
This PR:

* Paralelizes FFTLog.
* Paralelizes the spline integrator.
* Makes the latter accessible from python.

### CODE REVIEW

File: src/ccl_fftlog.c [hunk 1]
```diff
@@ -140,83 +141,126 @@ static void fht(int npk, int N,
...
+
+        #pragma omp for
+        for(int j = 0; j < npk; j++) {
+          for(int i = 0; i < N; i++)
+            a[i] = prefac_pk[i] * pk[j][i];
+
+          fftw_execute_dft(forward_plan,a,b);
+          for(int m = 0; m < N; m++)
+            b[m] *= u[m] / (double)(N);       // divide by N since FFTW doesn't normalize the inverse FFT
+          fftw_execute_dft(reverse_plan,b,b);
```
Comment (beckermr) [inline src/ccl_fftlog.c]: Is it safe to reuse plans from multiple threads?

Comment (damonge) [reply src/ccl_fftlog.c]: Yes (in fact having plans be initialized within parallel regions is a no-no)
http://www.fftw.org/fftw3_doc/Thread-safety.html

**Comment by damonge** [comment]
Right. So most of the FFTs we have to solve will be 1D and over vectors of O(1000) elements. In my experience, splitting these transforms over several cores only pays off when the vector is very large, when you have multi-dimensional transforms, or maybe when you only have a single transform.
If you have to solve several FFTs of the same size, I think doing each of them in a core in parallel is more efficient.

**Comment by damonge** [comment]
This will happen, for instance, for halo model power spectra (where you have to compute the Fourier profile for a large number of halo masses).

**Comment by beckermr** [comment]
Good enough for me!

**Comment by damonge** [comment]
Awesome!

**Pull request merged**